### DISCUSSION TITLE
Test units RunCommand and SshCommand.Result

### DISCUSSION THREAD

**Discussion by Pouet4205** [sshnet/SSH.NET#1452] [Q&A]
Hello, would it be possible to have a small unit test example for RunCommand and check SshCommand.Result and SshCommand.Error. I don't have much experience and I would like to improve my application using these tests

**Comment by Rob-Hague** [comment]
Have you seen the "Run a command" example on the front page? https://github.com/sshnet/SSH.NET

Or do you mean something else?

**Reply by Pouet4205** [reply]
I have no problem to use RunCommand, I use it in my function. I would like simulate the request and response of RunCommand to check my functions. It may need to use Mock but i dont know how to write these code

**Reply by Rob-Hague** [reply]
You could define your own interface and an implementing class which wraps the SshCommand object, then you can mock the interface. e.g.

```c#
interface ICommand
{
    string Execute();
}

class CommandWrapper : ICommand
{
    private SshCommand _command;

    public string Execute() => _command.Execute();
}
```
